### PR TITLE
Require JDK 23 to run Trino

### PR DESCRIPTION
**Pull request by wendigo**
Fixes https://github.com/trinodb/trino/issues/21316

## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
(x) Release notes are required, with the following suggested text:

```markdown
## General
* Require JDK 23 to run Trino ({issue}`21316`)
```

### CODE REVIEW

**Comment by wendigo** [comment]
cc @mosabua @martint

**Comment by mosabua** [comment]
Haha .. yeah .. I saw already @wendigo  .. personally I think we should get it merged soon .. maybe even for 462 ..

**Comment by wendigo** [comment]
The JDK 23.0.1 is due today @mosabua so once it's out, I'll update to it and we can consider merging it

**Comment by mosabua** [comment]
I will consider approving it then.. hahaha @wendigo

**Comment by wendigo** [comment]
@mosabua https://jdk.java.net/23/release-notes here you go :)

File: README.md [hunk 1]
```diff
@@ -54,7 +54,7 @@ Trino supports [reproducible builds](https://reproducible-builds.org) as of vers
 ## Build requirements
 
 * Mac OS X or Linux
-* Java 22.0.0+, 64-bit
+* Java 23.0.0+, 64-bit
```
Comment (mosabua) [inline README.md:57]: maybe ?? 

```suggestion
* Java 23.0.1+, 64-bit
```

Comment (martint) [reply README.md:57]: The current value is ok. The minimum required version for building and running is still 23.0.0. Unless we find a critical issue in a given minor version, there's no reason to bump the minimum requirement.

**Review by martint — APPROVED** [review]
Looks good, but what's going on with the tests?

File: docs/src/main/sphinx/installation/deployment.md [hunk 2]
```diff
@@ -32,10 +32,10 @@
 (requirements-java)=
 ### Java runtime environment
 
-Trino requires a 64-bit version of Java 22, with a minimum required version of
-22.0.1 and a recommendation to use the latest patch version. Earlier major versions
-such as Java 8, Java 11, Java 17 or Java 21 do not work. Newer versions such as
-Java 23 are not supported -- they may work, but are not tested.
+Trino requires a 64-bit version of Java 23, with a minimum required version of
+23.0.0 and a recommendation to use the latest patch version. Earlier versions
```
Comment (mosabua) [inline docs/src/main/sphinx/installation/deployment.md:36]: not 23.0.1 ?

**Comment by wendigo** [comment]
@martint ASM in the compiler needs to be updated since we've added java-module.info to the SPI

File: docs/src/main/sphinx/installation/deployment.md (hunk 2 above)
Comment (wendigo) [reply docs/src/main/sphinx/installation/deployment.md:36]: No, we updated to 23.0.1 but require 23.0.0 as a min version

File: README.md (hunk 1 above)
Comment (wendigo) [reply README.md:57]: Yep, that's my reasoning as well.

**Comment by wendigo** [comment]
Requires https://github.com/airlift/airbase/pull/546

**Comment by wendigo** [comment]
ppc64le is now available for 23.0.1 so this PR is ready to be merged @martint @mosabua

**Review by martint — APPROVED** [review]

**Review by mosabua — APPROVED** [review]
Let's ship it!

**Pull request merged**